src/bp/core/services/monitoring.ts

```diff
@@ -0,0 +1,99 @@
+import { injectable } from 'inversify'
+import _ from 'lodash'
+
+/**
+ * These methods are exposed outside the class to add minimal overhead when collecting metrics
+ * It also avoids possible circular reference (the logger class can't access monitoring service, for example)
+ */
+let metricCollectionEnabled = false
+let metricsContainer: MetricsContainer = {}
+
+export const resetMetrics = () => {
+  metricsContainer = {
+    'requests.count': 0,
+    'eventsIn.count': 0,
+    'eventsOut.count': 0,
+    'warnings.count': 0,
+    'errors.count': 0
+  }
+}
+
+export const getMetrics = () => {
+  const metrics: Partial<MonitoringMetrics> = {}
+  for (const pathMetric in metricsContainer) {
+    _.set(metrics, pathMetric, metricsContainer[pathMetric])
+  }
+
+  if (metrics.requests && metrics.requests.count && metrics.requests.latency_sum) {
+    metrics.requests.latency_avg = metrics.requests.latency_sum / metrics.requests.count
+  }
+
+  return metrics as MonitoringMetrics
+}
+
+export const setMetricsCollection = enabled => {
+  metricCollectionEnabled = enabled
+}
+
+export const incrementMetric = (metricName: string, value?: number | undefined) => {
```

Comment:
IMO Even this should not be available on CE. should simply return directly if on CE what do you think ?

Reply:
Since increments are all on methods in CE, I left that part there.  The metric collection is turned on on the pro side only, so checking for CE is useless.